webidls/Browser.webidl

```diff
@@ -0,0 +1,115 @@
+/**
+  * Browser is the equivalent of a top level Constellation::Frame.
+  * Browser is a top level browsing context.
+  * ~= Tab, ~= Webview
+  */
+
+[Constructor(DOMString contextName, StorageSession storageSession)]
+interface Browser {
+  // FIXME: how is this attached to a viewport? For swapping, it's important to be able to attach and detach
+
+  Promise<void> setHandler(BrowserHandler handler);
+
+  // Used to expose special JS APIs.
+  // For example, can be used to expose the Browser API or Web Extensions APIs for content
+  // See: https://developer.chrome.com/extensions/content_scripts#execution-environment
+  Promise<void> addJSModuleResolver(/*FIXME*/, boolean onlyForFrameScript);
+
+  void destroy();
+
+  // Can't change after browser creation
+  readonly attribute DOMString browsingContextName;
+  readonly attribute StorageSession storageSession;
+
+  // Will fail if entries already exist.
+  // This can be used for session restore, or to undo tab-close.
+  // On success, historyEntries is filled, active entry's pipeline is NOT pending.
+  // If necessary, other pipelines can be restored via HistoryEntry.restorePipeline()
+  Promise<void> restoreEntries(FrozenArray<LoadData> data, unsigned long activeIndex);
```

Comment:
That's not possible. You can't get HistoryEntry out of nowhere. Entries are constructed by a Browser.